mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import random
+from collections import Counter
+
+from mteb.abstasks import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+TEST_SAMPLES = 2048
+
+
+class TeluguAndhraJyotiNewsClassification(AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="TeluguAndhraJyotiNewsClassification",
+        description="A Telugu dataset for 5-class classification of Telugu news articles from 'Andhra Jyoti",
+        reference="https://github.com/AnushaMotamarri/Telugu-Newspaper-Article-Dataset",
+        dataset={
+            "path": "mlexplorer008/telugu_news_classification",
+            "revision": "3821aa93aa461c9263071e0897234e8d775ad616",
+        },
+        type="Classification",
+        category="s2s",
+        date=("2014-01-01", "2018-01-01"),
+        eval_splits=["test"],
+        eval_langs=["tel-Telu"],
+        main_score="accuracy",
+        form=["written"],
+        domains=["News"],
+        task_subtypes=["Topic classification"],
+        license="MIT",
+        socioeconomic_status="mixed",
+        annotations_creators="derived",
+        dialect=None,
+        text_creation="found",
+        bibtex_citation=None,
+        n_samples={"train": 17312, "test": TEST_SAMPLES},
+        avg_character_length={"train": 1435.53, "test": 1428.28},
+    )
+
+    def dataset_transform(self):
+        seed = 42
+        random.seed(seed)
+        self.dataset = self.dataset.rename_column("body", "text")
+        self.dataset = self.dataset.rename_column("topic", "label")
+
+        for split in ["test"]:
+            ds = self.dataset[split]
+            # Determine number of classes and samples per class
+            class_count = Counter([sample["label"] for sample in ds])
+            num_classes = len(class_count)
+            total_samples = min(TEST_SAMPLES, len(ds))
+            samples_per_class = total_samples // num_classes
+
+            # Try to maintain class balance
```

Comment:
this is great! I wonder if something like could help:
```
self.dataset = self.dataset["test"].train_test_split(
            test_size=TEST_SAMPLES, seed=42, stratify_by_column="label"
   ) 
```

Reply:
@isaac-chung  I don't think this is stratified sampling !. This is completely different.  label with least examples is maxed out, so that class imbalance is minimized

Reply:
I see, thanks!. @imenelydiaker and I are actually going through past datasets and we saw this as well in `VieStudentFeedbackClassification`. I wonder if we'd like to keep our stratified sampling strategy as default for all (to me this is sufficient), or adopt this strategy here.

Reply:
The stratification process is just here because of the subsampling, it allows us to keep the same samples a'd labels proportions as in the original dataset. Note that without the subsampling we would have run the evaluation on the whole dataset even if it's imbalanced.

So I think the question we should be asking is: do we want to have imbalanced classes during the evaluation? Because we're using KNN to evaluate the classification task, after that we compute an accuracy whatever the class balance is.

Reply:
Make sense. Having imbalanced classes are realistic and IMO inevitable. The main score for those datasets could be f1 instead of accuracy. Wdyt?

Reply:
Yes I think having f1 as a main score is a good idea! Should we update previous tasks that used accuracy?

Reply:
Yes if those datasets are noticeably imbalanced.